source/MiniMap.cpp

```diff
@@ -0,0 +1,287 @@
+		font.Draw(name, from + Point(MapPanel::OUTER, -.5 * font.Height()), lineColor);
+
+		// Draw the origin and destination systems, since they
+		// might not be linked via hyperspace.
+		Color color = Color(.5f * alpha, 0.f);
+		if(player.CanView(system) && system.IsInhabited(flagship) && gov)
+			color = gov->GetColor().Additive(alpha);
+		RingShader::Draw(from, MapPanel::OUTER, MapPanel::INNER, color);
+
+		// Add a circle around the system that the player is currently in.
+		if(&system == current)
+			RingShader::Draw(from, 11.f, 9.f, brightColor);
+
+		for(const System *link : system.Links())
+		{
+			// Only draw systems known to be attached to the jump systems.
+			if(!player.CanView(system) && !player.CanView(*link))
+				continue;
+
+			// Draw the system link. This will double-draw the jump
+			// path if it is via hyperlink, to increase brightness.
+			Point to = link->Position() - center + drawPos;
+			Point unit = (from - to).Unit() * MapPanel::LINK_OFFSET;
+			LineShader::Draw(from - unit, to + unit, MapPanel::LINK_WIDTH, lineColor);
+
+			if(drawnSystems.contains(link))
+				continue;
+			drawnSystems.insert(link);
+
+			gov = link->GetGovernment();
+			Color color = Color(.5f * alpha, 0.f);
+			if(player.CanView(*link) && link->IsInhabited(flagship) && gov)
+				color = gov->GetColor().Additive(alpha);
+			RingShader::Draw(to, MapPanel::OUTER, MapPanel::INNER, color);
+		}
+
+		unsigned missionCounter = 0;
+		for(const Mission &mission : player.Missions())
+		{
+			if(missionCounter >= MapPanel::MAX_MISSION_POINTERS_DRAWN)
+				break;
+
+			if(!mission.IsVisible())
+				continue;
+
+			if(mission.Destination()->IsInSystem(&system))
+			{
+				pair<bool, bool> blink = MapPanel::BlinkMissionIndicator(player, mission, step);
+				if(!blink.first)
+				{
+					bool isSatisfied = mission.IsSatisfied(player) && !mission.IsFailed() && blink.second;
+					MapPanel::DrawPointer(from, missionCounter, isSatisfied ? currentColor : blockedColor, false);
+				}
+				else
+					++missionCounter;
+			}
+
+			for(const System *waypoint : mission.Waypoints())
+			{
+				if(missionCounter >= MapPanel::MAX_MISSION_POINTERS_DRAWN)
+					break;
+				if(waypoint == &system)
+					MapPanel::DrawPointer(from, missionCounter, waypointColor, false);
+			}
+			for(const Planet *stopover : mission.Stopovers())
+			{
+				if(missionCounter >= MapPanel::MAX_MISSION_POINTERS_DRAWN)
+					break;
+				if(stopover->IsInSystem(&system))
+					MapPanel::DrawPointer(from, missionCounter, waypointColor, false);
+			}
+			for(const System *mark : mission.MarkedSystems())
+			{
+				if(missionCounter >= MapPanel::MAX_MISSION_POINTERS_DRAWN)
+					break;
+				if(mark == &system)
+					MapPanel::DrawPointer(from, missionCounter, waypointColor, false);
+			}
+		}
+	};
+
+	drawnSystems.insert(current);
+	if(target)
+		drawnSystems.insert(target);
+
+	drawSystemLinks(*current);
+	if(!target)
+		return;
```

Comment:
This is only checked when looping through the links, so we don't need `current` in the set when drawing it.
```suggestion
	if(target)
		drawnSystems.insert(target);
	drawSystemLinks(*current);
	if(!target)
		return;
	drawnSystems.insert(current);
```

Reply:
Do we really need to change this to save one (1) set insertion, though?